zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/tink/subtle/Ed25519Constants.kt

```diff
@@ -0,0 +1,1425 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+package app.cash.zipline.loader.internal.tink.subtle
+
+import app.cash.zipline.loader.internal.tink.subtle.Ed25519.CachedXYT
+
+/** Constants used in [Ed25519]. */
+internal object Ed25519Constants {
```

Comment:
This class is dramatically different from its predecessor. Rather than doing math at runtime I’m doing it all at build time.

Reply:
This was the program I used to create this file from its predecessor.

```
class DumpConstants {
  @Test
  fun dumpConstants() {
    println("val D = ${Ed25519Constants.D.toCode()}")
    println("val D2 = ${Ed25519Constants.D2.toCode()}")
    println("val SQRTM1 = ${Ed25519Constants.SQRTM1.toCode()}")
    println("val B_TABLE = ${Ed25519Constants.B_TABLE.toCode2()}")
    println("val B2 = ${Ed25519Constants.B2.toCode()}")
  }

  fun LongArray.toCode(): String {
    return joinToString(
      prefix="longArrayOf(",
      separator = ", ",
      postfix = ")",
    ) { "$it" }
  }

  internal fun List<List<Ed25519.CachedXYT>>.toCode2(): String {
    return joinToString (
      prefix="listOf(\n    ",
      separator = ",\n  ",
      postfix = "\n)",
    ) { it.toCode() }
  }

  internal fun List<Ed25519.CachedXYT>.toCode(): String {
    return joinToString (
      prefix="listOf(\n",
      separator = ",\n  ",
      postfix = ")",
    ) { it.toCode() }
  }

  internal fun Ed25519.CachedXYT.toCode(): String {
    return "CachedXYT(\n    yPlusX = ${yPlusX.toCode()},\n    yMinusX = ${yMinusX.toCode()},\n    t2d = ${t2d.toCode()}\n  )"
  }
}
```